### PR TITLE
add ernie_matching point-wise & pair-wise

### PR DESCRIPTION
**Pull request by tianxin1860**
### PR types

New features

### PR changes

Models

### Description

1. add single tower text_matching Point-wise based on ERNIE1.0 + Lcqmc
2. add single tower text_matching Pair-wise based on ERNIE1.0 + Lcqmc
3. Note: Point-wise model is evaluate by accuracy and Pair-wise model is evaluate by AUC metric

TODO:
add README.md

### CODE REVIEW

File: examples/text_matching/ernie_matching/data.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+        tmp_exmaples.append(example)
+
+        pool.append(example["title"])
+
+        if len(pool) >= pool_size:
+            np.random.shuffle(pool)
+            for idx, example in enumerate(tmp_exmaples):
+                example["neg_title"] = pool[idx]
+                #print("new_example:{}".format(example))
```
Comment (ZeyuChen) [inline examples/text_matching/ernie_matching/data.py]: 删掉冗余代码

File: examples/text_matching/ernie_matching/data.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        pool_size = len(dataset)
+
+    new_examples = []
+    pool = []
+    tmp_exmaples = []
+
+    for example in dataset:
+        label = example["label"]
+
+        # filter negative example
```
Comment (ZeyuChen) [inline examples/text_matching/ernie_matching/data.py]: 代码注释按照标准英文语法习惯，大写字母开头。

File: examples/text_matching/ernie_matching/model.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+import paddle.nn as nn
+import paddle.nn.functional as F
+
+
+class PonitWise(nn.Layer):
```
Comment (ZeyuChen) [inline examples/text_matching/ernie_matching/model.py]: PonitWise -> Pointwise
因为是一个单词，所以不用两个大写字母

File: examples/text_matching/ernie_matching/model.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                                    attention_mask)
+
+        cls_embedding = self.dropout(cls_embedding)
+        logits = self.classifier(cls_embedding)
+        probs = F.softmax(logits)
+
+        return probs
+
+
+class PairWise(nn.Layer):
```
Comment (ZeyuChen) [inline examples/text_matching/ernie_matching/model.py]: PairwiseMatching

File: examples/text_matching/ernie_matching/model.py (hunk 3 above)
Comment (ZeyuChen) [reply examples/text_matching/ernie_matching/model.py]: 我在想是用继承的方式增添XXXX.ForPairwiseMatching和XXXX.ForPointwiseMatching好

Comment (ZeyuChen) [reply examples/text_matching/ernie_matching/model.py]: 短期的话建议先调整下命名吧
PonitWise -> PointwiseMatching

**Review by ZeyuChen — COMMENTED** [review]
模型名称改为Pariwise/PointwiseMatching吧。
不然用形容词作为模型名不太合理

File: examples/text_matching/ernie_matching/data.py (hunk 1 above)
Comment (tianxin1860) [reply examples/text_matching/ernie_matching/data.py]: Done

File: examples/text_matching/ernie_matching/data.py (hunk 2 above)
Comment (tianxin1860) [reply examples/text_matching/ernie_matching/data.py]: 已修正

File: examples/text_matching/ernie_matching/model.py (hunk 4 above)
Comment (tianxin1860) [reply examples/text_matching/ernie_matching/model.py]: 已修正

File: examples/text_matching/ernie_matching/model.py (hunk 3 above)
Comment (tianxin1860) [reply examples/text_matching/ernie_matching/model.py]: 已经重命名，`XXXX.ForPairwiseMatching` 和 `XXXX.ForPointwiseMatching` 结构对预训练模型来说完全一致，后续可能需要评估下这部分逻辑放在 examples 下合适还是放在预训练模型的 `modeling.py` 中更合适。

File: examples/text_matching/ernie_matching/model.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+import paddle.nn as nn
+import paddle.nn.functional as F
+
+
+class PonitwiseMatching(nn.Layer):
```
Comment (ZeyuChen) [inline examples/text_matching/ernie_matching/model.py]: 应该是Point，不是Ponit

**Review by ZeyuChen — COMMENTED** [review]
PonitwiseMatching -> PointwiseMatching

File: examples/text_matching/ernie_matching/model.py (hunk 5 above)
Comment (tianxin1860) [reply examples/text_matching/ernie_matching/model.py]: 已修正，thanks.

**Review by ZeyuChen — APPROVED** [review]
LGTM
